src/modules/sv_detection_methods/analyze_sa_tag_method.cpp

```diff
@@ -57,29 +56,82 @@ void retrieve_aligned_segments(std::string const & sa_string, std::vector<Aligne
 void analyze_aligned_segments(std::vector<AlignedSegment> const & aligned_segments,
                               std::vector<Junction> & junctions,
                               seqan3::dna5_vector const & query_sequence,
-                              std::string const & read_name)
+                              std::string const & read_name,
+                              uint32_t const min_length,
+                              uint32_t const max_overlap)
 {
-    for(size_t i = 1; i<aligned_segments.size(); i++)
+    for (size_t i = 1; i<aligned_segments.size(); i++)
     {
         AlignedSegment current = aligned_segments[i-1];
         AlignedSegment next = aligned_segments[i];
         int32_t distance_on_read = next.get_query_start() - current.get_query_end();
-        // Check that there is neither an overlap nor a larger gap on the read
-        // TODO(eldarion): add command-line parameters for changing these currently hard-coded cutoffs
-        if (distance_on_read >= 0 && distance_on_read <= 10)
+        // Check that the overlap between two consecutive alignment segments
+        // of the read is lower than the given threshold
+        if (distance_on_read >= -static_cast<int32_t>(max_overlap))
```

Comment:
@SGSSGene: This line was the reason for my switch from `uint64_t` to `uint32_t`. I need to compare `distance_on_read` with `-max_overlap`. `distance_on_read` is an int32 while max_overlap was an uint64. To enable the comparison, I need to cast the unsigned integer to a signed integer. Casting `uint64_t` to `int32_t` seemed worse than casting `uint32_t` to `int32_t` even though this second conversion is still undefined for values larger than 2 billion. So I changed the parameter types from `uint64_t` to `uint32_t`.

Would it be a cleaner solution to cast the `uint32_t` to `int64_t` and compare this value to `distance_on_read`? If I interpret the standard correctly, then `distance_on_read` would be automatically converted to `int64_t`, too, for this comparison.